### ISSUE TITLE
TableView should render an endash with aria-label for cells with missing values

### ISSUE THREAD

**Issue by dkario** [adobe/react-spectrum#3263]
# 🙋 Feature Request

Spectrum recommends: "When there are gaps in the data, use an en dash (–) to represent null or not applicable (N/A) values." 
https://spectrum.adobe.com/page/table/#Use-an-en-dash-(%E2%80%93)-for-missing-values

It would be great if TableView code handled this internally so implementors don't have to. @reidbarber had the good idea to wrap the endash in `<span aria-label="No value">`, but I could see implementors either choosing different strings or not localizing it. I've seen many cases where implementors either don't know about the endash guidance or don't consider the aria-label (so it doesn't announce "endash"). 

## 🤔 Expected Behavior

If a TableView cell is empty, then a Cell is automatically rendered with an endash and a localized aria-label

## 😯 Current Behavior

Implementors have to manually insert the endash and a localized aria-label

## 💁 Possible Solution

Hopefully Cell code can handle this

## 🔦 Context

A Quarry contributor opened a PR with manually inserted endashes for empty fields, then @paulkenney recommended we figure out how AT should announce it

## 💻 Examples

https://git.corp.adobe.com/frontend/quarry/pull/2416/files#diff-ab1f0cc4f89901ddc21d0c244727e14f5d9713e9fd2417c560f3a43f7aa9bf7aR14-R16

## 🧢 Your Company/Team

Adobe/Quarry

**Comment by reidbarber** [comment]
This seems reasonable to me since Spectrum mentions this, and I can't think of any downsides. Maybe these cells can use a default string for an aria-label like `"No value"`, unless one is provided by the user.
